src/Persistence_SQL.php

```diff
@@ -687,63 +681,56 @@ public function load(Model $m, $id)
     }
 
     /**
-     * Tries to load data record, but will not fail if record can't be loaded.
+     * Loads a record from model and returns a associative array.
      *
      * @param Model $m
      * @param mixed $id
      *
      * @return array
      */
-    public function tryLoad(Model $m, $id)
+    public function load(Model $m, $id)
     {
-        $load = $this->action($m, 'select');
-        $load->where($m->getElement($m->id_field), $id);
-        $load->limit(1);
-
-        // execute action
-        $data = $load->getRow();
+        $data = $this->tryLoad($m, $id);
 
         if (!$data) {
-            $m->unload();
-
-            return [];
-        }
-
-        if (isset($data[$m->id_field])) {
-            $m->id = $data[$m->id_field];
-        } else {
             throw new Exception([
-                'ID of the record is unavailable. Read-only mode is not supported',
-                'model' => $m,
-                'id'    => $id,
-                'data'  => $data,
-            ]);
+                'Record was not found',
+                'model'      => $m,
+                'id'         => $id,
+                'conditions' => $m->conditions,
+            ], 404);
         }
 
         return $data;
     }
 
     /**
-     * Loads any one record.
+     * Tries to load any one record.
      *
      * @param Model $m
      *
      * @return array
      */
-    public function loadAny(Model $m)
+    public function tryLoadAny(Model $m)
```

Comment:
tryLoadAny can also use tryLoad($m) to avoid code duplication. That can be possible if in tryLoad method you make $id parameter optional and add condition:
```
if ($id !== null) $load->where($m->getElement($m->id_field), $id);
```